build/build.mk

```diff
@@ -23,6 +23,7 @@ LDFLAGS += -X main.gitCommit=${GIT_COMMIT}
 
 TEST_FLAGS += -failfast
 TEST_FLAGS += -race
+TEST_FLAGS += -ldflags '-X github.com/newrelic/infrastructure-agent/internal/integrations/v4/integration.minimumIntegrationIntervalOverride=2s'
```

Comment:
![](https://media.giphy.com/media/NEvPzZ8bd1V4Y/giphy.gif)

Reply:
clever one, yet not def convinced about it, nice topic for a coffee!

Reply:
the goal was not to expose that , since is our understanding that the min interval is not configurable, So we decided to do an small refactor so we can change that hardcoded value with the ldflag :)

Reply:
I think specialisation could be a good fit here to avoid [surprises](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) and won't expose any to be configured

anyway I really like reviewing this! food for though

we'll talk about this coffee in hand